base/MAPL_ISO8601_Time.F90

```diff
@@ -0,0 +1,1292 @@
+#include "MAPL_Exceptions.h"
+#include "MAPL_ErrLog.h"
+module MAPL_ISO8601_Time
+   use MAPL_KeywordEnforcerMod
+   use MAPL_ExceptionHandling
+   use ESMF
+   implicit none
+
+#ifndef DEBUG
+   private
+#endif
+
+   public :: convert_ISO8601_to_integer_time
+   public :: convert_ISO8601_to_integer_date
+   public :: convert_ISO8601_to_esmf_time
+   public :: convert_ISO8601_to_esmf_timeinterval
+
+   interface operator(.divides.)
+      module procedure :: divides
+   end interface
+
+   ! Error handling
+   integer, parameter :: INVALID = -1
+
+   ! parameters for processing date, time, and datetime strings
+   integer, parameter :: NUM_DATE_FIELDS = 3
+   integer, parameter :: NUM_TIME_FIELDS = 5
+   character(len=10), parameter :: DIGIT_CHARACTERS = '0123456789'
+   character, parameter :: TIME_PREFIX = 'T'
+   integer, parameter :: MINUTES_PER_HOUR = 60
+
+   integer, parameter :: Z = 0
+
+   integer, parameter :: ID_WIDTH = 100
+   integer, parameter :: ID_DAY = 1
+   integer, parameter :: ID_MONTH = ID_WIDTH * ID_DAY
+   integer, parameter :: ID_YEAR = ID_WIDTH * ID_MONTH
+
+   integer, parameter :: IT_WIDTH = ID_WIDTH
+   integer, parameter :: IT_SECOND = 1
+   integer, parameter :: IT_MINUTE = IT_WIDTH * IT_SECOND
+   integer, parameter :: IT_HOUR = IT_WIDTH * IT_MINUTE
+
+   type :: ISO8601Date
+#ifndef DEBUG
+      private
+#endif
+      integer :: year_
+      integer :: month_
+      integer :: day_
+   contains
+      procedure, public :: get_year
+      procedure, public :: get_month
+      procedure, public :: get_day
+   end type ISO8601Date
+
+   interface ISO8601Date
+      module procedure :: construct_ISO8601Date
+   end interface ISO8601Date
+
+   type :: date_fields
+      integer :: year_
+      integer :: month_
+      integer :: day_
+      logical :: is_valid_ = .FALSE.
+   end type date_fields
+
+   interface date_fields
+      module procedure :: construct_date_fields
+   end interface date_fields
+
+   type :: ISO8601Time
+#ifndef DEBUG
+      private
+#endif
+      integer :: hour_
+      integer :: minute_
+      integer :: second_
+      integer :: millisecond_
+      integer :: timezone_offset_ = Z
+   contains
+      procedure, public :: get_hour
+      procedure, public :: get_minute
+      procedure, public :: get_second
+      procedure, public :: get_millisecond
+      procedure, public :: get_timezone_offset
+   end type ISO8601Time
+
+   interface ISO8601Time
+      module procedure :: construct_ISO8601Time
+   end interface ISO8601Time
+
+   type :: time_fields
+      integer :: hour_
+      integer :: minute_
+      integer :: second_
+      integer :: millisecond_
+      integer :: timezone_offset_
+      logical :: is_valid_ = .FALSE.
+   end type time_fields
+
+   interface time_fields
+      module procedure :: construct_time_fields
+   end interface time_fields
+
+   type :: ISO8601DateTime
+#ifndef DEBUG
+      private
+#endif
+      type(ISO8601Date) :: date_
+      type(ISO8601Time) :: time_
```

Comment:
Looks like you probably do not need the trailing underscores now, as your accessors all have `get_` ...